exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/FederationGatewayService.java

```diff
@@ -23,14 +25,14 @@
 
     private final FederationGatewayClient client;
     private final DiagnosisKeyDao dd;
```

Comment:
I know I mentioned it before but I have to mention it again; abbreviating variable names is an anti-pattern, unless the abbreviation is clearer than the actual or an established name in itself. All code is clear to the writer, but the point is to be clear to someone else. Naming is there to communicate meaning and names like dd and fod communicate virtually nothing. IDEs are good with autocomplete so writing a proper name shouldn't be that much of a chore.

Reply:
Yep, this is on my not so urgent list. Not sure about anti-pattern though.

Reply:
E.g. in idea you could press <ctrl> to see the type od object if you are not sure. Too long names are just hard to read and makes code more messy.

Reply:
That applies in idea, but not when reviewing the code in github. It also doesn't apply to variables as a whole - only to singletons where the class and the role of the variable are 1-to-1. Often, repeating the classname isn't a very good name, but for these singletons it's also the role of the instance, so it works.

I wholeheartedly disagree that long names are hard to read and messy, unless by "too long" you mean really long names. No-one wants to see 80 char names and yes, shorter is better, but only if the shorter one communicates well. Abbreviations are not easier to read - they are mental mappings to the concept behind them, forcing the reader to do an extra cognitive step as he reads. Clear, descriptive names read naturally and eradicate the need to do that mapping.  Obviously, that's less of an issue when the mapped meaning is right next to the use, (local variable, loops, lambdas) but fields especially can live pretty far from the use-place and when you look at a diff, the declaration might not even be included.

When writing the code, you have it all in your mind anyhow, but a potential reader does not. He likely comes through multiple files/classes/functions on his way to understand some whole and doesn't know the abbreviations of this particular function by heart. On his way, he'll have to do an extra step to check the meaning of each variable, distracting from the real issue each time.

Reply:
Yep, I agree you have your point. I just have seen also these very long and hard to read namings. I'll make names more understandable after initial testings.